File: samples/petstore/php/tests/UserApiTest.php
<?php

require_once 'BaseApiTest.php';

class PetApiTest extends BaseApiTest {

  // Choose a random username for new pet insertion
  public static $randomUsername1;
  public static $randomUsername2;
  public static $randomUsername3;
  public static $randomUsername4;
  public static $randomUsername5;

  public static function setUpBeforeClass() {
    $letters = 'ABCDEFGHIJKLMNOPQRSTUVXYZ';
    $letter_arr = str_split($letters);
    shuffle($letter_arr);
    self::$randomUsername1 = implode('', $letter_arr);
    print "Username1 is " . self::$randomUsername1 . "\n\n";
    shuffle($letter_arr);
    self::$randomUsername2 = implode('', $letter_arr);
    print "Username2 is " . self::$randomUsername2 . "\n\n";
    shuffle($letter_arr);
    self::$randomUsername3 = implode('', $letter_arr);
    print "Username3 is " . self::$randomUsername3 . "\n\n";
    shuffle($letter_arr);
    self::$randomUsername4 = implode('', $letter_arr);
    print "Username4 is " . self::$randomUsername4 . "\n\n";
    shuffle($letter_arr);
    self::$randomUsername5 = implode('', $letter_arr);
    print "Username5 is " . self::$randomUsername5 . "\n\n";
  }

  public function testCreateUsersWithArrayInput() {

    $user = new User();

    $user->id = rand(10000, 100000);
    $user->lastName = rand(10000, 100000);
    $user->username = self::$randomUsername1;
    $user->phone = rand(10000, 100000);
    $user->email = rand(10000, 100000);
    $user->userStatus = rand(10000, 100000);
    $user->firstName = rand(10000, 100000);
    $user->password = rand(10000, 100000);

    $otherUser = new User();

    $otherUser->id = rand(10000, 100000);
    $otherUser->lastName = rand(10000, 100000);
    $otherUser->username = self::$randomUsername2;
    $otherUser->phone = rand(10000, 100000);
    $otherUser->email = rand(10000, 100000);
    $otherUser->userStatus = rand(10000, 100000);
    $otherUser->firstName = rand(10000, 100000);
    $otherUser->password = rand(10000, 100000);

    $users = array($user, $otherUser);
    $res = $this->userApi->createUsersWithArrayInput($users);

    $new_user = $this->userApi->getUserByName(self::$randomUsername1);

    $this->assertEquals($new_user->id, $user->id);
    $this->assertEquals($new_user->lastName, $user->lastName);
    $this->assertEquals($new_user->username, $user->username);
    $this->assertEquals($new_user->phone, $user->phone);
    $this->assertEquals($new_user->email, $user->email);
    $this->assertEquals($new_user->userStatus, $user->userStatus);
    $this->assertEquals($new_user->firstName, $user->firstName);
    $this->assertEquals($new_user->password, $user->password);

    $new_user = $this->userApi->getUserByName(self::$randomUsername2);

    $this->assertEquals($new_user->id, $otherUser->id);
    $this->assertEquals($new_user->lastName, $otherUser->lastName);
    $this->assertEquals($new_user->username, $otherUser->username);
    $this->assertEquals($new_user->phone, $otherUser->phone);
    $this->assertEquals($new_user->email, $otherUser->email);
    $this->assertEquals($new_user->userStatus, $otherUser->userStatus);
    $this->assertEquals($new_user->firstName, $otherUser->firstName);
    $this->assertEquals($new_user->password, $otherUser->password);

  }

  public function testCreateUsersWithListInput() {

    $user = new User();

    $user->id = rand(10000, 100000);
    $user->lastName = rand(10000, 100000);
    $user->username = self::$randomUsername4;
    $user->phone = rand(10000, 100000);
    $user->email = rand(10000, 100000);
    $user->userStatus = rand(10000, 100000);
    $user->firstName = rand(10000, 100000);
    $user->password = rand(10000, 100000);

    $otherUser = new User();

    $otherUser->id = rand(10000, 100000);
    $otherUser->lastName = rand(10000, 100000);
    $otherUser->username = self::$randomUsername5;
    $otherUser->phone = rand(10000, 100000);
    $otherUser->email = rand(10000, 100000);
    $otherUser->userStatus = rand(10000, 100000);
    $otherUser->firstName = rand(10000, 100000);
    $otherUser->password = rand(10000, 100000);

    $users = array($user, $otherUser);
    $res = $this->userApi->createUsersWithArrayInput($users);

    $new_user = $this->userApi->getUserByName(self::$randomUsername4);

    $this->assertEquals($new_user->id, $user->id);
    $this->assertEquals($new_user->lastName, $user->lastName);
    $this->assertEquals($new_user->username, $user->username);
    $this->assertEquals($new_user->phone, $user->phone);
    $this->assertEquals($new_user->email, $user->email);
    $this->assertEquals($new_user->userStatus, $user->userStatus);
    $this->assertEquals($new_user->firstName, $user->firstName);
    $this->assertEquals($new_user->password, $user->password);

    $new_user = $this->userApi->getUserByName(self::$randomUsername5);

    $this->assertEquals($new_user->id, $otherUser->id);
    $this->assertEquals($new_user->lastName, $otherUser->lastName);
    $this->assertEquals($new_user->username, $otherUser->username);
    $this->assertEquals($new_user->phone, $otherUser->phone);
    $this->assertEquals($new_user->email, $otherUser->email);
    $this->assertEquals($new_user->userStatus, $otherUser->userStatus);
    $this->assertEquals($new_user->firstName, $otherUser->firstName);
    $this->assertEquals($new_user->password, $otherUser->password);

  }

  public function testCreateUser() {

    $user = new User();

    $user->id = rand(10000, 100000);
    $user->lastName = rand(10000, 100000);
    $user->username = self::$randomUsername3;
    $user->phone = rand(10000, 100000);
    $user->email = rand(10000, 100000);
    $user->userStatus = rand(10000, 100000);
    $user->firstName = rand(10000, 100000);
    $user->password = rand(10000, 100000);

    $res = $this->userApi->createUser($user);

    $new_user = $this->userApi->getUserByName(self::$randomUsername3);

    $this->assertEquals($new_user->id, $user->id);
    $this->assertEquals($new_user->lastName, $user->lastName);
    $this->assertEquals($new_user->username, $user->username);
    $this->assertEquals($new_user->phone, $user->phone);
    $this->assertEquals($new_user->email, $user->email);
    $this->assertEquals($new_user->userStatus, $user->userStatus);
    $this->assertEquals($new_user->firstName, $user->firstName);
    $this->assertEquals($new_user->password, $user->password);

  }

  public function testUpdateUser() {

    $user = $this->userApi->getUserByName(self::$randomUsername1);

    $user->lastName = rand(10000, 100000);
    $user->phone = rand(10000, 100000);
    $user->email = rand(10000, 100000);
    $user->userStatus = rand(10000, 100000);
    $user->firstName = rand(10000, 100000);
    $user->password = rand(10000, 100000);

    $res = $this->userApi->updateUser(self::$randomUsername1, $user);

    $updated_user = $this->userApi->getUserByName(self::$randomUsername1);

    $this->assertEquals($updated_user->lastName, $user->lastName);
    $this->assertEquals($updated_user->username, $user->username);
    $this->assertEquals($updated_user->phone, $user->phone);
    $this->assertEquals($updated_user->email, $user->email);
    $this->assertEquals($updated_user->userStatus, $user->userStatus);
    $this->assertEquals($updated_user->firstName, $user->firstName);
    $this->assertEquals($updated_user->password, $user->password);

  }

  public function testDeleteUser() {

    $res = $this->userApi->deleteUser(self::$randomUsername1);

    $deleted_user = $this->userApi->getUserByName(self::$randomUsername1);

    $this->assertEquals($deleted_user, null);

  }

  public function testLoginUser() {

    $res = $this->userApi->loginUser("anyusername", "anypassword");

    $this->assertEquals(substr($res, 0, 23), "logged in user session:");

  }

  public function testLogoutUser() {

    $res = $this->userApi->logoutUser();

    // We just want to make sure there are no errors in this test.
    // To verify you are getting back a 200, you might want to add
    // something like this to Swagger.php callAPI():
    // print "Response for call to $resourcePath : ";
    // print_r($data);

  }

}
?>

File: samples/petstore/python3/tests/UserApiTest.py
#!/usr/bin/env python
import sys
import unittest
import random
import urllib

from BaseApiTest import BaseApiTest

sys.path = ['./'] + sys.path
from petstore import *
from petstore.models import *


def randomString():
    return str(randomInt())


def randomInt():
    return random.randint(1000, 100000)


class UserApiTest(BaseApiTest):

    @classmethod
    def setUpClass(cls):
        alpahbet = list('ABCDEFGHIJKLMNOPQRSTUVWXYZ')
        random.shuffle(alpahbet)
        cls.randomUsername1 = ''.join(alpahbet)
        random.shuffle(alpahbet)
        cls.randomUsername2 = ''.join(alpahbet)
        random.shuffle(alpahbet)
        cls.randomUsername3 = ''.join(alpahbet)
        random.shuffle(alpahbet)
        cls.randomUsername4 = ''.join(alpahbet)
        random.shuffle(alpahbet)
        cls.randomUsername5 = ''.join(alpahbet)

    def testCreateUsersWithArrayInput(self):

        user = User.User()

        user.id = randomInt()
        user.lastName = randomString()
        user.username = self.randomUsername1
        user.phone = randomString()
        user.email = randomString()
        user.userStatus = int(randomString())
        user.firstName = randomString()
        user.password = randomString()

        otherUser = User.User()

        otherUser.id = randomInt()
        otherUser.lastName = randomString()
        otherUser.username = self.randomUsername2
        otherUser.phone = randomString()
        otherUser.email = randomString()
        otherUser.userStatus = int(randomString())
        otherUser.firstName = randomString()
        otherUser.password = randomString()

        users = [user, otherUser]
        self.userApi.createUsersWithArrayInput(users)

        newUser = self.userApi.getUserByName(self.randomUsername1)

        assert newUser.id == user.id, 'id matches user'
        assert newUser.lastName == user.lastName, 'lastName matches user'
        assert newUser.username == user.username, 'username matches user'
        assert newUser.phone == user.phone, 'phone matches user'
        assert newUser.email == user.email, 'email matches user'
        assert newUser.userStatus == user.userStatus, 'status matches user'
        assert newUser.firstName == user.firstName, 'firstName matches user'
        assert newUser.password == user.password, 'password matches user'

        newUser = self.userApi.getUserByName(self.randomUsername2)

        assert newUser.id == otherUser.id, 'id matches user'
        assert newUser.lastName == otherUser.lastName, 'lastName matches user'
        assert newUser.username == otherUser.username, 'username matches user'
        assert newUser.phone == otherUser.phone, 'phone matches user'
        assert newUser.email == otherUser.email, 'email matches user'
        assert newUser.userStatus == otherUser.userStatus, 'status matches user'
        assert newUser.firstName == otherUser.firstName, 'firstName matches user'
        assert newUser.password == otherUser.password, 'password matches user'

    def testCreateUsersWithListInput(self):

        user = User.User()

        user.id = randomInt()
        user.lastName = randomString()
        user.username = self.randomUsername3
        user.phone = randomString()
        user.email = randomString()
        user.userStatus = int(randomString())
        user.firstName = randomString()
        user.password = randomString()

        otherUser = User.User()

        otherUser.id = randomInt()
        otherUser.lastName = randomString()
        otherUser.username = self.randomUsername4
        otherUser.phone = randomString()
        otherUser.email = randomString()
        otherUser.userStatus = int(randomString())
        otherUser.firstName = randomString()
        otherUser.password = randomString()

        users = [user, otherUser]
        self.userApi.createUsersWithListInput(users)

        newUser = self.userApi.getUserByName(self.randomUsername3)

        assert newUser.id == user.id, 'id matches user'
        assert newUser.lastName == user.lastName, 'lastName matches user'
        assert newUser.username == user.username, 'username matches user'
        assert newUser.phone == user.phone, 'phone matches user'
        assert newUser.email == user.email, 'email matches user'
        assert newUser.userStatus == user.userStatus, 'status matches user'
        assert newUser.firstName == user.firstName, 'firstName matches user'
        assert newUser.password == user.password, 'password matches user'

        newUser = self.userApi.getUserByName(self.randomUsername4)

        assert newUser.id == otherUser.id, 'id matches user'
        assert newUser.lastName == otherUser.lastName, 'lastName matches user'
        assert newUser.username == otherUser.username, 'username matches user'
        assert newUser.phone == otherUser.phone, 'phone matches user'
        assert newUser.email == otherUser.email, 'email matches user'
        assert newUser.userStatus == otherUser.userStatus, 'status matches user'
        assert newUser.firstName == otherUser.firstName, 'firstName matches user'
        assert newUser.password == otherUser.password, 'password matches user'

    def testCreateUser(self):

        user = User.User()

        user.id = randomInt()
        user.lastName = randomString()
        user.username = self.randomUsername5
        user.phone = randomString()
        user.email = randomString()
        user.userStatus = int(randomString())
        user.firstName = randomString()
        user.password = randomString()

        self.userApi.createUser(user)

        newUser = self.userApi.getUserByName(self.randomUsername5)

        assert newUser.id, user.id
        assert newUser.lastName, user.lastName
        assert newUser.username, user.username
        assert newUser.phone, user.phone
        assert newUser.email, user.email
        assert newUser.userStatus, user.userStatus
        assert newUser.firstName, user.firstName
        assert newUser.password, user.password

    def testUpdateUser(self):

        user = User.User()
        username = randomString()
        user.id = randomInt()
        user.lastName = randomString()
        user.username = username
        user.phone = randomString()
        user.email = randomString()
        user.userStatus = int(randomString())
        user.firstName = randomString()
        user.password = randomString()

        self.userApi.createUser(user)

        user = self.userApi.getUserByName(username)

        user.lastName = randomString()
        user.phone = randomString()
        user.email = randomString()
        user.userStatus = int(randomString())
        user.firstName = randomString()
        user.password = randomString()

        self.userApi.updateUser(username, user)

        updatedUser = self.userApi.getUserByName(username)

        assert updatedUser.lastName == user.lastName, 'should match lastName'
        assert updatedUser.username == user.username, 'should match username'
        assert updatedUser.phone == user.phone, 'should match phone'
        assert updatedUser.email == user.email, 'should match email'
        assert updatedUser.userStatus == user.userStatus, 'should match status'
        assert updatedUser.firstName == user.firstName, 'should match firstName'
        assert updatedUser.password == user.password, 'should match password'

    def testDeleteUser(self):

        user = User.User()
        username = randomString()
        user.username = username

        self.userApi.createUser(user)

        self.userApi.deleteUser(username)

        userGone = False

        try:
            self.userApi.getUserByName(username)
        except urllib.request.HTTPError:
            userGone = True

        assert userGone, 'user should be deleted'

    def testLoginUser(self):

        res = self.userApi.loginUser("anyusername", "anypassword")

        assert res[:23] == "logged in user session:", 'should get session'

    def testLogoutUser(self):
        # We just want to make sure there are no errors in this test.
        self.userApi.logoutUser()


if __name__ == "__main__":
    unittest.main()
